### DISCUSSION TITLE
error.api.internal_server_error: Unauthorized #3815

### DISCUSSION THREAD

**Discussion by pinkygangula** [chaos-mesh/chaos-mesh#3828] [Q&A]
Randomly getting error.api.internal_server_error: Unauthorized while i'm using chaos-mesh dashboard , to solve this issue creating RBAC token every time like below:

![Uploading Screen Shot 2022-11-30 at 8.45.39 AM.png…]()

kubernates version:v4.5.7
chaos-mesh version: v2.4.2

**Comment by fewdan** [comment]
@pinkygangula 
Can you provide version information as follows?
`kubectl version`
The result is like this:
`
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.11", GitCommit:"27522a29febbcc4badac257763044d0d90c11abd", GitTreeState:"clean", BuildDate:"2021-09-15T19:21:44Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.12", GitCommit:"696a9fdd2a58340e61e0d815c5769d266fca0802", GitTreeState:"clean", BuildDate:"2022-04-13T19:01:10Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
`
It is possible that you are your new version of k8s, and the expiration time of the token is one hour.

**Reply by pinkygangula** [reply]
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:28:30Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.12-gke.100", GitCommit:"fd16f64da04c784b85909576a3f7abf4ed49b949", GitTreeState:"clean", BuildDate:"2022-09-22T09:23:33Z", GoVersion:"go1.17.13b7", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.23) exceeds the supported minor version skew of +/-1

**Reply by fewdan** [reply]
> Versions of Kubernetes before v1.22 automatically created long term credentials for accessing the Kubernetes API. This older mechanism was based on creating token Secrets that could then be mounted into running Pods. In more recent versions, including Kubernetes v1.25, API credentials are obtained directly by using the TokenRequest API, and are mounted into Pods using a projected volume. The tokens obtained using this method have bounded lifetimes, and are automatically invalidated when the Pod they are mounted into is deleted.
> You can still manually create a service account token Secret; for example, if you need a token that never expires. However, using the TokenRequest subresource to obtain a token to access the API is recommended instead.

@pinkygangula 
The above paragraph was copied from the official kubernetes documentation. According to the [official documentation](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#manual-secret-management-for-serviceaccounts), the token will expire.

So I guess, it is because your token has expired after exceeding the life cycle, so you can refer to the [this document](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-long-lived-api-token-for-a-serviceaccount) to create a never-expiring token and try it out.

If it is not caused by token expiration, we will explore other possibilities.
